code/particle/ParticleManager.cpp

```diff
@@ -0,0 +1,336 @@
+#include <algorithm>
+#include <memory>
+
+#include "particle/ParticleManager.h"
+
+#include "particle/effects/SingleParticleEffect.h"
+#include "particle/effects/CompositeEffect.h"
+#include "particle/effects/ConeGeneratorEffect.h"
+
+#include "bmpman/bmpman.h"
+#include "globalincs/systemvars.h"
+
+/**
+ * @defgroup particleSystems Particle System
+ */
+
+namespace {
+using namespace particle;
+
+const char* effectTypeNames[static_cast<size_t>(EffectType::MAX)] = {
+	"Single",
+	"Composite",
+	"Cone"
+};
+
+const char* getEffectTypeName(EffectType type) {
+	Assertion(static_cast<size_t>(type) >= static_cast<size_t>(EffectType::Single)
+				  && static_cast<size_t>(type) < static_cast<size_t>(EffectType::MAX),
+			  "Invalid effect type specified!");
+
+	return effectTypeNames[static_cast<size_t>(type)];
+}
+
+ParticleEffectPtr constructEffect(const SCP_string& name, EffectType type) {
+	using namespace effects;
+	// Use an unique_ptr to make sure memory is deallocated if an exception is thrown
+	std::unique_ptr<ParticleEffect> effect;
+
+	switch (type) {
+		case EffectType::Single: {
+			effect.reset(new SingleParticleEffect(name));
+			effect->parseValues(false);
+			break;
+		}
+		case EffectType::Composite: {
+			effect.reset(new CompositeEffect(name));
+			effect->parseValues(false);
+			break;
+		}
+		case EffectType::Cone: {
+			effect.reset(new ConeGeneratorEffect(name));
+			effect->parseValues(false);
+			break;
+		}
+		default: {
+			Error(LOCATION, "Unimplemented effect type %d encountered! Get a coder!", static_cast<int>(type));
+			throw std::runtime_error("Unimplemented effect type encountered!");
+		}
+	}
+
+	return effect.release();
+}
+
+EffectType parseEffectType() {
+	required_string("$Type:");
+
+	SCP_string type;
+	stuff_string(type, F_NAME);
+
+	for (size_t i = 0; i < static_cast<size_t>(EffectType::MAX); ++i) {
+		if (!stricmp(type.c_str(), effectTypeNames[i])) {
+			return static_cast<EffectType>(i);
+		}
+	}
+
+	error_display(0, "Unknown effect type '%s'!", type.c_str());
+	return EffectType::Invalid;
+}
+
+void parseCallback(const char* fileName) {
+	using namespace particle;
+
+	read_file_text(fileName, CF_TYPE_TABLES);
+
+	reset_parse();
+
+	required_string("#Particle Effects");
+
+	while (optional_string("$Effect:")) {
+		SCP_string name;
+		stuff_string(name, F_NAME);
+
+		auto type = parseEffectType();
+
+		ParticleManager::get()->addEffect(constructEffect(name, type));
+	}
+
+	required_string("#End");
+}
+
+void parseConfigFiles() {
+	parse_modular_table("*-part.tbm", parseCallback);
```

Comment:
Why should there be a main table even if it isn't used by anyone? I guess it helps the documentation but at least for the scripting.tbl case it has caused some problems where different mods have conflicting tables.

Adding a table is only a one line change but I don't see a good reason why a table should exist.